lib/matplotlib/tests/test_colors.py

```diff
@@ -769,13 +769,33 @@ def test_conversions():
         hex_color
 
 
+def test_to_rgba_array_single_str():
+    # single color name is valid
+    assert_array_equal(mcolors.to_rgba_array("red"), [(1, 0, 0, 1)])
+
+    # single char color sequence is deprecated
+    with pytest.warns(cbook.MatplotlibDeprecationWarning) as record:
+        array = mcolors.to_rgba_array("rgb")
+    assert len(record) == 1
+    assert ("Using a string of single character colors as a color sequence "
+            "is deprecated" in record[0].message.args[0])
+    assert_array_equal(array, [(1, 0, 0, 1), (0, 0.5, 0, 1), (0, 0, 1, 1)])
+
+    # invalid string
+    with pytest.raises(ValueError) as e:
```

Comment:
`pytest.raises(..., match=...)`

Reply:
pytest is really imprecise here. Because https://docs.pytest.org/en/latest/reference.html?highlight=pytest.raises#pytest-raises states
> match – if specified, a string containing a regular expression, or a regular expression object, that is tested against the string representation of the exception using re.match.

I didn't want to bother escaping the plain string. But plain strings appear to match as well. :roll_eyes: